File: src/main/java/com/hellsepontus/commands/logic/ChangePropCommand.java
package com.hellsepontus.commands.logic;

import com.hellsepontus.commands.CastCommand;
import com.hellsepontus.commands.Command;

import java.util.Map;

public class ChangePropCommand extends CastCommand{
    
    public static final String ID = "changeprop";
    
    public ChangePropCommand() {
        super(ID);
    }

    @Override
    public void execute() {
        Command responseCmd = Command.createCommand(ChangePropApplyCommand.ID, this);
        responseCmd.execute();
    }

    public static class ChangePropApplyCommand extends CastCommand{

        public static final String ID = "changeprop-apply";

        public ChangePropApplyCommand() {
            super(ID);
        }

        @Override
        public void execute() {
            sendToAll();
        }

        @Override
        public Map<String, Object> toData() {
            return this.data;
        }
    }
}

File: public/js/game/commands/logic/changePropCommand.js
function ChangePropCommand(uid, propName, newValue){
    Command.call(this, ChangePropCommand.ID);
    this.toData = function(){
        return {uid: uid, pn: propName, val: newValue};
    }
}
ChangePropCommand.ID = 'changeprop'
Command.add(ChangePropCommand.ID, ChangePropCommand)

function ChangePropApplyCommand(){
    Command.call(this, ChangePropApplyCommand.ID);
    this.data = true
    this.execute = function(){
        var gameUser = Model.room.findUserById(this.data.uid)
        if(this.data.pn == 'health'){
            gameUser.addHealth(this.data.val - gameUser.health, true)
        }else if(this.data.pn == 'score'){
            gameUser.addScore(this.data.val - gameUser.score, true)
        }
    }
}
ChangePropApplyCommand.ID = 'changeprop-apply'
Command.add(ChangePropApplyCommand.ID, ChangePropApplyCommand)